server/data/data-jmap/src/main/java/org/apache/james/jmap/api/model/UploadId.java

```diff
@@ -0,0 +1,81 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.api.model;
+
+import org.apache.commons.text.RandomStringGenerator;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.hash.Hashing;
+
+public class UploadId {
+    public static final RandomStringGenerator RANDOM_STRING_GENERATOR = new RandomStringGenerator.Builder()
+        .withinRange('a', 'z')
+        .build();
+
+    public static UploadId random() {
+        return new UploadId(RANDOM_STRING_GENERATOR.generate(20));
+    }
+
+    public static UploadId from(String id) {
+        Preconditions.checkNotNull(id);
+        Preconditions.checkArgument(!id.isEmpty());
+        return new UploadId(id);
+    }
+
+    public static UploadId from(byte[] bytes) {
+        Preconditions.checkNotNull(bytes);
+        return new UploadId(Hashing.sha256().hashBytes(bytes).toString());
```

Comment:
We should remove this: hashing and dedup is a blob store concern. We should only use the `random()` method in our generation logic.